### DISCUSSION TITLE
How to use rust-analyzer?

### DISCUSSION THREAD

**Discussion by IowU** [pulsar-edit/.github#214] [Community Package Help] [answered]
Hello,
I was looking in the packages if [rust-analyzer](https://rust-analyzer.github.io) was present, but couldn't find it.
Is it available as a different name?
I also have the rust-analyzer binary on my PC: is there a way to tell Pulsar to use it for rust files? Also, is there a way to configure it?

**Comment by confused-Techie** [comment]
Keep in mind all those packages are community built, so if you can't find what you want please feel free to make it.

Otherwise just searching "rust" does bring back quite a [few](https://web.pulsar-edit.dev/packages/search?q=rust) results, so it's very possible one or a combination of these do exactly what you want.

I'd test a few out and see if any of them work. I can't say if any use `rust-analyzer` but if that tool is popular it's very likely one or more of them do

**Comment by Daeraxa** [comment, marked as answer]
Rust-analyzer is "just" an language server, it doesn't do anything by itself. The [ide-rust](https://web.pulsar-edit.dev/packages/ide-rust) package is made by the rust-lang org specifically for Atom to work with rust-analyzer (which they also maintain).

I think the generic-lsp package under development also works with rust-analyzer now too.

**Reply by Spiker985** [reply]
I can confirm that generic-lsp does work for `rust-analyzer` and I've gotten it to work with `powershell-editor-services` as well

**Reply by IowU** [reply]
I installed the `ide-rust` package and now I'm seeing the hints from `rust-analyzer` as I expected.
Thank you for your help.

**Reply by Cheukting** [reply]
@Daeraxa thank you for pointing me to this thread. However, it does not work for me. I cannot install the ide-rust package, here is the error message:
```
No native build was found for platform=darwin arch=arm64 runtime=electron abi=87 uv=1 armv=8 libc=glibc node=14.16.0 electron=12.2.3
```
I guess they are not maintaining it anymore and there is no new build that will match my environment.
